xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp

```diff
@@ -125,7 +125,14 @@ bool CVideoPlayerVideo::OpenStream(CDVDStreamInfo hint)
   if (hint.flags & AV_DISPOSITION_ATTACHED_PIC)
     return false;
   if (hint.extrasize == 0)
-    return false;
+  {
+    // codecs with no extradata
+    if (hint.codec != AV_CODEC_ID_MJPEG &&
```

Comment:
@Rechi could you reduce extradata check to those codecs only:
AV_CODEC_ID_MPEG1VIDEO
AV_CODEC_ID_MPEG2VIDEO
AV_CODEC_ID_MPEG2VIDEO_XVMC
AV_CODEC_ID_H264
AV_CODEC_ID_HEVC
AV_CODEC_ID_MPEG4
AV_CODEC_ID_WMV3
AV_CODEC_ID_VC1

others are ok without extradata